new dataWriter for bodies

vancouver29 opened the pull request:
(mass, position(x,y), velocity(dx,dy))
this function is based on your data_write() function

vancouver29 committed:
new dataWriter for bodies

(mas, position(x,y), velocity(dx,dy))

johnxnguyen commented on src/util/mod.rs:
```diff
@@ -44,6 +45,25 @@ impl DataWriter {
+
+    /// Creates a new file in the writers directory with each body written on a separate line.
+    pub fn write_body_file(&mut self, bodies: Vec<Body>) {
+        let path = format!("{}/frame-{}.txt", self.directory, self.counter);
```
The idea to write the body data into a file is to be able to save the initial state of the simulation. That means we only need to create one single file. I think you're confusing the need to write frames with point data with the need to save the list of bodies.

johnxnguyen reviewed (changes requested):
Please adjust your code so that the new method accepts a list of bodies and creates a single file. The method could also take an argument for the file name.

vancouver29 committed:
Body writer

bodies are written in a file

johnxnguyen commented on src/util/mod.rs:
```diff
@@ -46,6 +47,34 @@ impl DataWriter {
+        }
+    }
+
+    pub fn write_bodies(&self, bodies: Vec<Body> , path: String) -> std::io::Result<()> {
```
The argument `path` is a bit misleading. It's being used as a file name, so it would be better to call it `filename` or something like that.

johnxnguyen commented on src/util/mod.rs:
```diff
@@ -46,6 +47,34 @@ impl DataWriter {
+        let path = format!("{}/{}.txt", self.directory, path);
+        let mut file = fs::File::create(path)?;
+        for body in bodies {
+            write!(file, "{}-{}-{}-{}-{}\n", body.mass, body.position.x, body.position.y,
```
Formatting the body data with dashes (`-`) is a bit odd and makes it hard to read. I would suggest to separate values with a comma (`,`) like we do when writing points.

johnxnguyen commented on src/util/mod.rs:
```diff
@@ -66,15 +95,34 @@ mod tests {
+        body_writer.write_bodies(vec![Body::new(2.1, Point::new(1.1,2.2),
+                                                Vector::new(3.4, 4.5))], "bodies".to_owned());
+//        body_writer.write_bodies(vec![Body::new(3.0, Point::new(4.0,5.0),
+//                                                Vector::zero())], "bodies".to_owned());
```
Commented out code, you should remove it.

johnxnguyen commented on src/util/mod.rs:
```diff
@@ -66,15 +95,34 @@ mod tests {
+        let mut lines = String::new();
+        file.read_to_string(&mut lines);
+        assert_eq!(lines, "2.1-1.1-2.2-3.4-4.5\n".to_owned());
+    }
```
Also don't forget to remove the directory at the end of the test, this keeps the project folder clean.

johnxnguyen committed:
Merge branch 'develop' into feature/dataWriter-bodies

vancouver29 committed:
Update write_bodies function

1. parameter "path"  is now changed to file_name
2. seperated symbol for the bodies in file is now changed to "," instead of "-"
3. Commented out code is now removed
4. Directory of that test is also removed after the test

vancouver29 committed:
path is changed to file_name

vancouver29 committed:
"-" is changed to ","

vancouver29 committed:
commented out code is now removed

vancouver29 committed:
the directory for test is also removed after the test
